doc/http-backoff.md

```diff
@@ -0,0 +1,10 @@
+# HTTP retry and backoff logic
+
+When Ignition is fetching a resource over http(s), if the resource is unavailable Ignition will continually retry to fetch the resource with an exponential backoff between requests. Once enough time has passed, Ignition will cancel the attempt and fail.
+
+There are two timeout values that can be configured with an Ignition config:
+
+- `httpResponseHeaders`: this field controls how long Ignition will wait for the server to send initial response headers for a given request attempt. If there is an outstanding request for a resource for which there has been no response, it will be cancelled when this timeout is reached. Ignition will then wait for a period of time and then try again. A value of 0 indicates that there is no timeout, and the default is 10 seconds.
```

Comment:
I like the style of the schema doc more, where we use a bullet list to show keys hierarchy.